hack/update-mocks.sh

```diff
@@ -30,6 +30,7 @@ _tmp="${KUBE_ROOT}/_tmp_build_tag_files"
 mkdir -p "${_tmp}"
 
 function cleanup {
+    go clean -modcache
```

Comment:
I think this is only safe if we move registering the trap to _after_ `kube::golang::setup_env` otherwise we won't have set GOPATH etc and we may wind up calling `go clean` on the system-wide modcache instead if exit happens before setup_env

Reply:
also, is this actually specific to this script? are we installing things under _tmp? if this is about _output, I don't think this script should be wiping it.

Reply:
Thanks for reviewing @BenTheElder . Happy to investigate this. I'll take a look.

Reply:
figured out what was up with the modcache in https://github.com/kubernetes/kubernetes/pull/117526